lib/extract/dl_esm_inf/netcdf/kernel_data_netcdf.f90

```diff
@@ -0,0 +1,448 @@
+        use netcdf
+        use field_mod, only : r2d_field
+        implicit none
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        type(r2d_field), intent(in) :: value
+        character(*), intent(in), optional :: prefix
+        integer :: x_dimid, y_dimid, retval
+        integer, dimension(2) :: dimids
+        character(128) :: real_prefix
+
+        if (present(prefix)) then
+            real_prefix = prefix
+        else
+            real_prefix = ""
+        endif
+        retval = CheckError(nf90_def_dim(this%ncid, name//"dim1",  &
+                                         value%grid%nx, x_dimid))
+        retval = CheckError(nf90_def_dim(this%ncid, name//"dim2",  &
+                                         value%grid%ny, y_dimid))
+        dimids =  (/ x_dimid, y_dimid /)
+        retval = CheckError(nf90_def_var(this%ncid, trim(name//real_prefix), Nf90_REAL8,     &
+                                         dimids, this%var_id(this%next_var_index)))
+
+        this%next_var_index = this%next_var_index + 1
+    end subroutine DeclareFieldDouble
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine writes the value of a dl_esm_field (r2d_field)
+    !! to the NetCDF file. It takes the variable id from the corresponding
+    !! declaration.
+    !! @param[inout] this The instance of the PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine WriteFieldDouble(this, name, value)
+        use netcdf
+        use field_mod, only : r2d_field
+        implicit none
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        type(r2d_field), intent(in) :: value
+        integer :: retval
+        retval = CheckError(nf90_put_var(this%ncid, this%var_id(this%next_var_index),  &
+                                         value%data(:,:)))
+        this%next_var_index = this%next_var_index + 1
+    end subroutine WriteFieldDouble
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine reads the values of a dl_esm_inf field (r2d_field).
+    !! It allocates a 2d-double precision array to store the read values
+    !! which is then returned to the caller. If the memory for the array can
+    !! not be allocated, the application will be stopped.
+    !! @param[inout] this The instance of the PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[out] value An allocatable, unallocated 2d-double precision array
+    !!             which is allocated here and stores the values read.
+    subroutine ReadFieldDouble(this, name, value)
+        use netcdf
+        implicit none
+
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: name
+        double precision, dimension(:,:), allocatable :: value
```

Comment:
Intent(out) (or does it have to be inout)?

Reply:
I believe out is indeed correct, which I used.